File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        
        mapping = {} 
        rev_mapping = {}

        if(len(s)!=len(t)):
            return False

        for i in range(len(t)):

            a=s[i]
            b=t[i]

            if a in mapping:
                if mapping.get(a)!=b:
                    return False
            else:
                if b in rev_mapping:
                    return False
                else:
                    mapping[a] = b
                    rev_mapping[b]=a

        print(mapping)
        return True

File: 0205-isomorphic-strings/0205-isomorphic-strings.scala
import scala.collection.mutable
object Solution {
    def isIsomorphic(s: String, t: String): Boolean = {

        var mapping = mutable.HashMap[Char, Char]()
        var revMapping = mutable.HashMap[Char, Char]()
        var output = true
        for(idx<-t.indices){
            val a = s(idx)
            val b = t(idx)

            if(mapping.contains(a)){
                mapping.get(a) match {
                    case Some(mappedChar) => if (mappedChar != b) {
                    output = false
                    }
                    case _ => {}
                }
            }
            else{
                if(revMapping.contains(b)){
                    output=false
                }
                mapping.put(a,b)
                revMapping.put(b,a)
            }
        }
        output
    }
}